frontend/src/views/Lobby.tsx

```diff
@@ -337,19 +343,64 @@ function LobbyPage() {
       conditionallyBootKickedUser(room, currentUser);
     };
 
+    setLoading(true);
     connect(roomId, userId).then(() => {
       // Body encrypt through JSON.
       subscribe(routes(roomId).subscribe_lobby, subscribeCallback).then((subscriptionParam) => {
         setSubscription(subscriptionParam);
         setSocketConnected(true);
+        setError('');
+        setLoading(false);
       }).catch((err) => {
         setError(err.message);
+        setLoading(false);
       });
     }).catch((err) => {
       setError(err.message);
+      setLoading(false);
     });
   }, [currentUser, conditionallyBootKickedUser]);
 
+  const refreshRoomDetails = () => {
+    // Call GET endpoint to get latest room info
+    if (!loading) {
+      setLoading(true);
+      getRoom(location.state.roomId)
+        .then((res) => {
+          setStateFromRoom(res);
+
+          // Boot the user from the room, if they are not present.
+          updateCurrentUserDetails(res.users);
+
+          // Attempt to connect the user to the socket.
+          if (currentUser && currentUser.userId) {
+            connectUserToRoom(res.roomId, currentUser.userId);
```

Comment:
Are there any checks for when the user is already connected not to connect them again?

Reply:
I believe that check is done in the actual [`connect` api function](https://github.com/rocketden/main/blob/1bffb0d706f386b57ac51f001e8e62853386276c/frontend/src/api/Socket.ts#L36).